core/controllers/profile.py

```diff
@@ -317,30 +317,30 @@ def put(self) -> None:
         update_type = self.payload.get('update_type')
         data = self.payload.get('data')
         bulk_email_signup_message_should_be_shown = False
+        user_settings = user_services.get_user_settings(self.user_id)
```

Comment:
Just a note, it's still fine to do some lightweight validation in the controller as you did before. This is because the validation here is for the incoming dict, not for the model, and it is good practice in general to prevent bad data getting into the domain layer. So if you think that's a good idea then feel free to go for it.

Reply:
Done.
Yes, it's good idea.
I have thought to do it using continue statements in the 'for' loop in the next PR, so that there is no need to repeat validation for string many times.